src/lndk_offers.rs

```diff
@@ -48,6 +62,53 @@ pub fn decode(offer_str: String) -> Result<Offer, Bolt12ParseError> {
     offer_str.parse::<Offer>()
 }
 
+// Checks that the user-provided amount matches the offer.
+pub async fn validate_amount(
```

Comment:
There's def some other validation we could do besides the amount... But I'd prefer getting the main offers logic into place first before we get too deep in the weeds there.